### ISSUE TITLE
The release process is regenerating binaries. 

### ISSUE THREAD

**Issue by Vad1mo** [goharbor/harbor-cli#230]
The current release process is using goreleaser to make the binaries, on the other hand when we create the images for the harbor-cli, we are rebuilding the binaries. IMO it would make sense to reuse the binaries instead of regenerating them. 

- same process 
- same signatures and hashes of generated binaries.

**Comment by bupd** [comment]
It seems we can use goreleaser to create docker images too. 
- https://goreleaser.com/customization/docker/
- https://goreleaser.com/customization/sbom/

**Comment by bupd** [comment]
Also we can use goreleaser to publish to AUR and homebrew too.

**Comment by bupd** [comment]
I am brainstorming here.

insteading building binaries inside dagger,  We can update the Release pipeline to use the binaries generated by the goreleaser and package them into containers created by dagger. so we can use the same sbom. too

**Comment by bupd** [comment]
@danish9039 
yes, create binaries and sbom with goreleaser, and publish to github release.

for the dagger part, we should use the binary from goreleaser. and attach the sbom to the container.

**Comment by danish9039** [comment]
@bupd ,
I’ve completed the initial implementation for this PR and would really appreciate your review and any guidance you can share.
Please let me know if there’s anything I should improve or if I’ve missed something. Thanks in advance!
